fdbserver/DataDistribution.actor.cpp

```diff
@@ -755,68 +755,45 @@ struct DDTeamCollection : ReferenceCounted<DDTeamCollection> {
 
 			int64_t bestLoadBytes = 0;
 			Optional<Reference<IDataDistributionTeam>> bestOption;
-			std::vector<std::pair<int, Reference<IDataDistributionTeam>>> randomTeams;
-			std::set< UID > sources;
-
+			std::vector<Reference<IDataDistributionTeam>> randomTeams;
+			std::set<UID> completeSources;
+			for( int i = 0; i < req.completeSources.size(); i++ ) {
+				completeSources.insert( req.completeSources[i] );
+			}
 			if( !req.wantsNewServers ) {
-				std::vector<Reference<IDataDistributionTeam>> similarTeams;
-				bool foundExact = false;
-
-				for( int i = 0; i < req.sources.size(); i++ )
-					sources.insert( req.sources[i] );
-
-				for( int i = 0; i < req.sources.size(); i++ ) {
-					if( self->server_info.count( req.sources[i] ) ) {
-						auto& teamList = self->server_info[ req.sources[i] ]->teams;
+				int bestSize = 0;
+				for( int i = 0; i < req.completeSources.size(); i++ ) {
+					if( self->server_info.count( req.completeSources[i] ) ) {
```

Comment:
I think changing the if to 
`if( !self->server_info.count( req.completeSources[i] ) ) {continue;}` is more intuitive.

The purpose of the for-loop is only to find the first server of `req.completeSources` that is in `server_info`. It does not loop back once it finds the first such server.